carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/core/capability/CapabilitiesBuilder.java

```diff
@@ -0,0 +1,113 @@
+            break;
+        case EDGE:
+            EdgeCapabilities edgeCapabilities = new EdgeCapabilities();
+            capabilities = this.customCapabilities == null ? edgeCapabilities.getCapabilities()
+                    : edgeCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case GECKO:
+            GeckoCapabilities geckoCapabilities = new GeckoCapabilities();
+            capabilities = this.customCapabilities == null ? geckoCapabilities.getCapabilities()
+                    : geckoCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case FIREFOX:
+            FirefoxCapabilities firefoxCapabilities = new FirefoxCapabilities();
+            capabilities = this.customCapabilities == null ? firefoxCapabilities.getCapabilities()
+                    : firefoxCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case ANDROID:
+            UIAutomator2Capabilities uiAutomator2Capabilities = new UIAutomator2Capabilities();
+            capabilities = this.customCapabilities == null ? uiAutomator2Capabilities.getCapabilities()
+                    : uiAutomator2Capabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case IOS:
+            XCUITestCapabilities xcuiTestCapabilities = new XCUITestCapabilities();
+            capabilities = this.customCapabilities == null ? xcuiTestCapabilities.getCapabilities()
+                    : xcuiTestCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case WINDOWS:
+            WindowsCapabilities windowsCapabilities = new WindowsCapabilities();
+            capabilities = this.customCapabilities == null ? windowsCapabilities.getCapabilities()
+                    : windowsCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case SAFARI:
+            SafariCapabilities safariCapabilities = new SafariCapabilities();
+            capabilities = this.customCapabilities == null ? safariCapabilities.getCapabilities()
+                    : safariCapabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        case MAC:
+            Mac2Capabilities mac2Capabilities = new Mac2Capabilities();
+            capabilities = this.customCapabilities == null ? mac2Capabilities.getCapabilities()
+                    : mac2Capabilities.createCapabilitiesFromCustom(this.customCapabilities);
+            break;
+        }
+
+        if (capabilities == null) {
+            throw new RuntimeException("Cannot understand what type of driver should be chosen. Please, provide browser name or automation name");
+        }
+
+        if (!Objects.equals(this.testName, StringUtils.EMPTY)) {
+            capabilities.setCapability("name", testName);
```

Comment:
I saw "name" capability init higher I guess, why 2 times?

Reply:
No, this capability added only one time - and only for chrome, firefox and edge

Reply:
This class removed